src/main/java/com/rtbhouse/kafka/workers/impl/task/TaskManager.java

```diff
@@ -103,7 +111,8 @@ private void stopThreads() throws InterruptedException {
         // waits for all threads to stop because only then tasks can be rebalanced safely
         synchronized (rebalanceLock) {
             while (!allThreadsNotRunning()) {
-                rebalanceLock.wait();
+                rebalanceLock.wait(1_000L);
```

Comment:
Would 1 second be enough?

Reply:
We don't need to be super accurate in throwing timeout errors.